MekHQ/src/mekhq/campaign/parts/equipment/LargeCraftAmmoBin.java

```diff
@@ -270,9 +270,14 @@ public MissingPart getMissingPart() {
 
     @Override
     public void updateConditionFromEntity(boolean checkForDestruction) {
+        final String METHOD_NAME = "updateConditionFromEntity()"; //$NON-NLS-1$
         if(null != unit) {
             Mounted mounted = unit.getEntity().getEquipment(equipmentNum);
-            if(null != mounted) {
+            if (!(mounted.getType() instanceof AmmoType)) {
+                MekHQ.getLogger().log(LargeCraftAmmoBin.class, METHOD_NAME, LogLevel.WARNING,
+                        unit.getName() + " has invalid ammo bin at equipment number " + equipmentNum);
+            }
+            if(null != mounted && mounted.getType() instanceof AmmoType) {
```

Comment:
You definitely need a null check on mounted before calling mounted.getType(). I'd suggest moving the logic from the if statement on line 280 into the if statement on line 276, then making 280 an "else".